src/sign-verify/clu_dgst_setup.c

```diff
@@ -57,33 +60,125 @@ static void wolfCLU_dgstHelp(void)
     WOLFCLU_LOG(WOLFCLU_L0, "Parameters:");
     WOLFCLU_LOG(WOLFCLU_L0, "\t-signature file containing the signature");
     WOLFCLU_LOG(WOLFCLU_L0, "\t-verify key used to verify the signature");
+    WOLFCLU_LOG(WOLFCLU_L0, "\t-sign   private key used to create the signature");
+    WOLFCLU_LOG(WOLFCLU_L0, "\t-out    output file for signature");
     WOLFCLU_LOG(WOLFCLU_L0, "Example:");
     WOLFCLU_LOG(WOLFCLU_L0, "\twolfssl dgst -signature test.sig -verify key.pem test");
 }
 
 
+/* return WOLFCLU_SUCCESS on success */
+static int ExtractKey(void* key, WOLFSSL_EVP_PKEY* pkey, int* keySz,
+        enum wc_SignatureType* sigType, int signing)
+{
+    ecc_key* ecc = NULL;
+    RsaKey*  rsa = NULL;
+    byte* der = NULL;
+    int   derSz = 0;
+    int ret = WOLFCLU_SUCCESS;
+    word32 idx = 0;
+
+    if (signing == 0) { /* expecting public key */
+        derSz = wolfCLU_pKeytoPubKey(pkey, &der);
+    }
+    else { /* expecting private key */
+        derSz = wolfCLU_pKeytoPriKey(pkey, &der);
+    }
+    if (derSz <= 0) {
+        WOLFCLU_LOG(WOLFCLU_E0, "Unable to extract der key");
+        ret = WOLFCLU_FATAL_ERROR;
+    }
+
+    switch (wolfSSL_EVP_PKEY_id(pkey)) {
+        case EVP_PKEY_RSA:
+            *keySz   = (int)sizeof(RsaKey);
+            *sigType = WC_SIGNATURE_TYPE_RSA_W_ENC;
+            rsa = (RsaKey*)key;
+
+            if (wc_InitRsaKey(rsa, NULL) != 0) {
```

Comment:
Not calling `wc_FreeRsaKey`

Reply:
ecc and rsa free get called later on around line 451 in the file. The caller makes use of rsa/ecc and does the free'ing.